src/yoroi-wallets/cardano/utxoManager.ts

```diff
@@ -75,3 +77,61 @@ export const makeUtxoStorage = (storage: typeof storageLegacy, storagePath: stri
 
   return utxoStorage
 }
+
+const makeUtxoManagerStorage = (storage: typeof storageLegacy, storagePath: string) => {
+  const addrCounterPath = `${storagePath}/addrCounter`
+  return {
+    addrCounter: {
+      save: (addrCounter: number) => storage.write(addrCounterPath, addrCounter),
+      clear: async () => {
+        await storage.remove(addrCounterPath)
+      },
+      read: async () => (await storage.read<number>(addrCounterPath)) ?? 0,
+    },
+  } as const
+}
+
+export const makeUtxoManager = async (id: string, apiUrl: string, storage: typeof storageLegacy) => {
+  const managerStorage = makeUtxoManagerStorage(storage, `/wallet/${id}/utxosManager`)
+  const serviceStorage = makeUtxoStorage(storage, `/wallet/${id}/utxos`)
+  const service = initUtxo(serviceStorage, `${apiUrl}/`)
+
+  let addrCounter = await managerStorage.addrCounter.read()
+
+  const getCachedUtxos = () => service.getAvailableUtxos().then((utxos) => utxos.map(serializer))
+  const initialUtxos = await getCachedUtxos()
+
+  // utxo state is related to the addresses used, if it changes a reset is needed
+  const sync = async (addresses: Array<string>) => {
+    if (addresses.length === addrCounter) return service.syncUtxoState(addresses)
```

Comment:
it doesn't need to check the addresses, they are always created in blocks of 50 and always increase by 50. the utxos state from lib is derived from the addresses but it's not saved (can't know, it looks like more a client responsibility), so if we change the addresses we need to wipe the lib utxo service state

Reply:
if is the same number of address, it just syncs
if is not the same number -> wipe, sync, store